.github/workflows/e2etest-run-modulestest.yml

```diff
@@ -0,0 +1,92 @@
+name: Run Tests
+on:
+  workflow_call:
+    inputs:
+      environment:
+        description: The name of the target environment
+        type: string
+        required: true
+      distroName:
+        description: The name of the target distro
+        type: string
+        required: true
+      resourceGroupName:
+        description: The name of the target environment
+        type: string
+        required: true
+      buildType:
+        description: The name of the build type
+        type: string
+        default: Release
+        required: false
+      repeatForXHours:
+        description: Repeat the test suite for X hours, every test run gets a new test-report
+        type: number
+        required: false
+      testNameSuffix:
+        description: The name to use for the suffix of the test report
+        type: string
+        required: false
+
+jobs:
+  run-tests:
+    name: ${{ matrix.distroName }}
+    environment: ${{ inputs.environment }}
+    strategy:
+      fail-fast: false
+      matrix:
+        distroName: ${{ fromJson(inputs.distroName) }}
+    runs-on: ${{ inputs.resourceGroupName }}-${{ matrix.distroName }}
+    timeout-minutes: 1440
+    steps:
+      - name: Checkout code
+        run: |
+          git clone https://github.com/Azure/azure-osconfig --branch $GITHUB_REF_NAME --recursive .
```

Comment:
`actions/checkout@3` instead of `git clone ...`?

```YAML
- uses: actions/checkout@v3
  with:
    submodules: recursive
    ref: $GITHUB_REF_NAME # should be able to omit this since the action defaults to the reference or SHA for the triggering event
```

https://github.com/actions/checkout

Reply:
I did try that but the `actions/checkout@v3` it requires a version of git not available in debian9, this way worked across all distros for the recursive checkout